File: src/main/java/com/fyp/studentinterfaceservice/services/interfaces/CompanyService.java
package com.fyp.studentinterfaceservice.services.interfaces;

import com.fyp.studentinterfaceservice.dto.CompanyWrapper;
import com.fyp.studentinterfaceservice.model.Company;
import com.fyp.studentinterfaceservice.model.CompanyProfile;
import com.fyp.studentinterfaceservice.model.MailingList;
import com.fyp.studentinterfaceservice.model.Review;

import java.util.List;

public interface CompanyService {
    CompanyWrapper getCompany(String name);


    List<Company> getAllCompanies();

    Review review(Review review);

    double getRating(String name);

    CompanyProfile updateProfile(CompanyProfile profile);

    MailingList addToMailingList(Long companyId, String email);

    MailingList getMailingList(Long companyId);
}

File: src/main/resources/student-interface-service/src/app/service/company.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {CompanyProfileModel} from "../model/company-profile.model";

@Injectable({
  providedIn: 'root'
})
export class CompanyService {

  constructor(private http: HttpClient) { }

  getCompany(name: string): Observable<any> {
    return this.http.get('http://localhost:8083/companies/' + name);
  }

  getAllCompanies(): Observable<any> {
    return this.http.get('http://localhost:8083/companies/all');
  }

  getRating(name: string): Observable<any> {
    return this.http.get('http://localhost:8083/companies/rating/' + name);
  }

  updateCompanyProfile(profile : CompanyProfileModel): Observable<any> {
    return this.http.put('http://localhost:8083/companies/updateProfile', {
      "profileId": profile.profileId,
      "hiredStudents": profile.hiredStudents,
      "reviews": profile.reviews
    });
  }

  addToMailingList(companyId: number, email:string): Observable<any> {
    return this.http.post('http://localhost:8083/companies/addToMailing/' + companyId, {},
      {params: {email: email}})
  }
}
